diff --git a/node_modules/bitcoinjs-lib/LICENSE b/node_modules/bitcoinjs-lib/LICENSE
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/README.md b/node_modules/bitcoinjs-lib/README.md
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/address.d.ts b/node_modules/bitcoinjs-lib/src/address.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/address.js b/node_modules/bitcoinjs-lib/src/address.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/bip66.d.ts b/node_modules/bitcoinjs-lib/src/bip66.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/bip66.js b/node_modules/bitcoinjs-lib/src/bip66.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/block.d.ts b/node_modules/bitcoinjs-lib/src/block.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/block.js b/node_modules/bitcoinjs-lib/src/block.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/bufferutils.d.ts b/node_modules/bitcoinjs-lib/src/bufferutils.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/bufferutils.js b/node_modules/bitcoinjs-lib/src/bufferutils.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/crypto.d.ts b/node_modules/bitcoinjs-lib/src/crypto.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/crypto.js b/node_modules/bitcoinjs-lib/src/crypto.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/ecc_lib.d.ts b/node_modules/bitcoinjs-lib/src/ecc_lib.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/ecc_lib.js b/node_modules/bitcoinjs-lib/src/ecc_lib.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/index.d.ts b/node_modules/bitcoinjs-lib/src/index.d.ts
old mode 100644
new mode 100755
index 420979f..12beaf8
--- a/node_modules/bitcoinjs-lib/src/index.d.ts
+++ b/node_modules/bitcoinjs-lib/src/index.d.ts
@@ -3,10 +3,11 @@ import * as crypto from './crypto';
 import * as networks from './networks';
 import * as payments from './payments';
 import * as script from './script';
-export { address, crypto, networks, payments, script };
+import * as scriptSig from './script_signature';
+export { address, crypto, networks, payments, script, scriptSig };
 export { Block } from './block';
 export { TaggedHashPrefix } from './crypto';
-export { Psbt, PsbtTxInput, PsbtTxOutput, Signer, SignerAsync, HDSigner, HDSignerAsync, } from './psbt';
+export { Psbt, PsbtTxInput, PsbtTxOutput, Signer, SignerAsync, HDSigner, HDSignerAsync, getHashForSig } from './psbt';
 export { OPS as opcodes } from './ops';
 export { Transaction } from './transaction';
 export { Network } from './networks';
diff --git a/node_modules/bitcoinjs-lib/src/index.js b/node_modules/bitcoinjs-lib/src/index.js
old mode 100644
new mode 100755
index 4159064..34a07cf
--- a/node_modules/bitcoinjs-lib/src/index.js
+++ b/node_modules/bitcoinjs-lib/src/index.js
@@ -10,6 +10,8 @@ exports.initEccLib =
   exports.networks =
   exports.crypto =
   exports.address =
+  exports.scriptSig =
+  exports.getHashForSig =
     void 0;
 const address = require('./address');
 exports.address = address;
@@ -22,6 +24,8 @@ exports.payments = payments;
 const script = require('./script');
 exports.script = script;
 var block_1 = require('./block');
+const scriptSig = require('./script_signature');
+exports.scriptSig = scriptSig;
 Object.defineProperty(exports, 'Block', {
   enumerable: true,
   get: function () {
@@ -35,6 +39,7 @@ Object.defineProperty(exports, 'Psbt', {
     return psbt_1.Psbt;
   },
 });
+exports.getHashForSig = psbt_1.getHashForSig;
 var ops_1 = require('./ops');
 Object.defineProperty(exports, 'opcodes', {
   enumerable: true,
diff --git a/node_modules/bitcoinjs-lib/src/merkle.d.ts b/node_modules/bitcoinjs-lib/src/merkle.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/merkle.js b/node_modules/bitcoinjs-lib/src/merkle.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/networks.d.ts b/node_modules/bitcoinjs-lib/src/networks.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/networks.js b/node_modules/bitcoinjs-lib/src/networks.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/ops.d.ts b/node_modules/bitcoinjs-lib/src/ops.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/ops.js b/node_modules/bitcoinjs-lib/src/ops.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/bip341.d.ts b/node_modules/bitcoinjs-lib/src/payments/bip341.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/bip341.js b/node_modules/bitcoinjs-lib/src/payments/bip341.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/embed.d.ts b/node_modules/bitcoinjs-lib/src/payments/embed.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/embed.js b/node_modules/bitcoinjs-lib/src/payments/embed.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/index.d.ts b/node_modules/bitcoinjs-lib/src/payments/index.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/index.js b/node_modules/bitcoinjs-lib/src/payments/index.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/lazy.d.ts b/node_modules/bitcoinjs-lib/src/payments/lazy.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/lazy.js b/node_modules/bitcoinjs-lib/src/payments/lazy.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2ms.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2ms.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2ms.js b/node_modules/bitcoinjs-lib/src/payments/p2ms.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2pk.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2pk.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2pk.js b/node_modules/bitcoinjs-lib/src/payments/p2pk.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2pkh.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2pkh.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2pkh.js b/node_modules/bitcoinjs-lib/src/payments/p2pkh.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2sh.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2sh.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2sh.js b/node_modules/bitcoinjs-lib/src/payments/p2sh.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2tr.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2tr.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2tr.js b/node_modules/bitcoinjs-lib/src/payments/p2tr.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2wpkh.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2wpkh.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2wpkh.js b/node_modules/bitcoinjs-lib/src/payments/p2wpkh.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2wsh.d.ts b/node_modules/bitcoinjs-lib/src/payments/p2wsh.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/payments/p2wsh.js b/node_modules/bitcoinjs-lib/src/payments/p2wsh.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/psbt.d.ts b/node_modules/bitcoinjs-lib/src/psbt.d.ts
old mode 100644
new mode 100755
index de7bbb3..b42319f
--- a/node_modules/bitcoinjs-lib/src/psbt.d.ts
+++ b/node_modules/bitcoinjs-lib/src/psbt.d.ts
@@ -119,6 +119,17 @@ export declare class Psbt {
     addUnknownKeyValToOutput(outputIndex: number, keyVal: KeyValue): this;
     clearFinalizedInput(inputIndex: number): this;
 }
+interface PsbtCache {
+    __NON_WITNESS_UTXO_TX_CACHE: Transaction[];
+    __NON_WITNESS_UTXO_BUF_CACHE: Buffer[];
+    __TX_IN_CACHE: { [index: string]: number };
+    __TX: Transaction;
+    __FEE_RATE?: number;
+    __FEE?: number;
+    __EXTRACTED_TX?: Transaction;
+    __UNSAFE_SIGN_NONSEGWIT: boolean;
+}  
+export declare function getHashForSig(inputIndex: number, input: PsbtInput, cache: PsbtCache, forValidate: boolean, sighashTypes: number[]);
 interface PsbtOptsOptional {
     network?: Network;
     maximumFeeRate?: number;
diff --git a/node_modules/bitcoinjs-lib/src/psbt.js b/node_modules/bitcoinjs-lib/src/psbt.js
old mode 100644
new mode 100755
index 9aa3668..91d5e33
--- a/node_modules/bitcoinjs-lib/src/psbt.js
+++ b/node_modules/bitcoinjs-lib/src/psbt.js
@@ -1,6 +1,6 @@
 'use strict';
 Object.defineProperty(exports, '__esModule', { value: true });
-exports.Psbt = void 0;
+exports.Psbt = exports.getHashForSig = void 0;
 const bip174_1 = require('bip174');
 const varuint = require('bip174/src/lib/converter/varint');
 const utils_1 = require('bip174/src/lib/utils');
@@ -1277,6 +1277,7 @@ function getHashForSig(inputIndex, input, cache, forValidate, sighashTypes) {
     hash,
   };
 }
+exports.getHashForSig = getHashForSig;
 function getAllTaprootHashesForSig(inputIndex, input, inputs, cache) {
   const allPublicKeys = [];
   if (input.tapInternalKey) {
diff --git a/node_modules/bitcoinjs-lib/src/psbt/bip371.d.ts b/node_modules/bitcoinjs-lib/src/psbt/bip371.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/psbt/bip371.js b/node_modules/bitcoinjs-lib/src/psbt/bip371.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/psbt/psbtutils.d.ts b/node_modules/bitcoinjs-lib/src/psbt/psbtutils.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/psbt/psbtutils.js b/node_modules/bitcoinjs-lib/src/psbt/psbtutils.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/push_data.d.ts b/node_modules/bitcoinjs-lib/src/push_data.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/push_data.js b/node_modules/bitcoinjs-lib/src/push_data.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script.d.ts b/node_modules/bitcoinjs-lib/src/script.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script.js b/node_modules/bitcoinjs-lib/src/script.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script_number.d.ts b/node_modules/bitcoinjs-lib/src/script_number.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script_number.js b/node_modules/bitcoinjs-lib/src/script_number.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script_signature.d.ts b/node_modules/bitcoinjs-lib/src/script_signature.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/script_signature.js b/node_modules/bitcoinjs-lib/src/script_signature.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/transaction.d.ts b/node_modules/bitcoinjs-lib/src/transaction.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/transaction.js b/node_modules/bitcoinjs-lib/src/transaction.js
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/types.d.ts b/node_modules/bitcoinjs-lib/src/types.d.ts
old mode 100644
new mode 100755
diff --git a/node_modules/bitcoinjs-lib/src/types.js b/node_modules/bitcoinjs-lib/src/types.js
old mode 100644
new mode 100755
